### PR TITLE
Changed DataTable, Diagram, Image, Menu, Meter, TableBody, TableCell, Tabs

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable, Diagram, Image, Menu, Meter, TableBody, TableCell, Tabs to remove hocs and fix some missing forwardRefs.

#### Where should the reviewer start?

anywhere

#### What testing has been done on this PR?

unit

#### How should this be manually tested?

storybook

#### Any background context you want to provide?

cleaning up some more, eventually we'll be able to get rid of all higher order components and the `recompose` dependency.

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Cell.js [hunk 1]
```diff
@@ -25,8 +23,8 @@ const Cell = ({
   primaryProperty,
   rowProp,
   scope,
-  theme,
 }) => {
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/DataTable/Cell.js:27]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```
I think we decided to be consistent with defaultProps fallback.

File: src/js/components/DataTable/ExpanderCell.js [hunk 2]
```diff
@@ -10,7 +8,8 @@ import { Button } from '../Button';
 import { TableCell } from '../TableCell';
 import { normalizeColor } from '../../utils';
 
-const ExpanderCell = ({ context, expanded, onToggle, theme, ...rest }) => {
+const ExpanderCell = ({ context, expanded, onToggle, ...rest }) => {
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/DataTable/ExpanderCell.js:12]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```

File: src/js/components/DataTable/Header.js [hunk 3]
```diff
@@ -29,10 +28,10 @@ const Header = ({
   onToggle,
   pad,
   sort,
-  theme,
   widths,
   ...rest
 }) => {
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/DataTable/Header.js:34]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```

File: src/js/components/DataTable/Resizer.js [hunk 4]
```diff
@@ -9,7 +14,8 @@ const ResizerBox = styled(Box)`
   cursor: col-resize;
 `;
 
-const Resizer = ({ onResize, property, theme }) => {
+const Resizer = ({ onResize, property }) => {
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/DataTable/Resizer.js:18]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```

File: src/js/components/DataTable/Searcher.js [hunk 5]
```diff
@@ -15,14 +12,8 @@ import { Text } from '../Text';
...
-const Searcher = ({
-  filtering,
-  filters,
-  onFilter,
-  onFiltering,
-  property,
-  theme,
-}) => {
+const Searcher = ({ filtering, filters, onFilter, onFiltering, property }) => {
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/DataTable/Searcher.js:16]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```

File: src/js/components/DataTable/Sorter.js [hunk 6]
```diff
@@ -19,9 +18,9 @@ const Sorter = ({
   onSort,
   property,
   sort,
-  theme,
   themeProps,
 }) => {
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/DataTable/Sorter.js:23]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```

File: src/js/components/Image/Image.js [hunk 7]
```diff
@@ -1,46 +1,39 @@
...
-      {...extraProps}
-      ref={forwardRef}
-      opacityProp={opacity}
-      fillProp={fill}
-      src={!imageMissing ? src : fallback}
-    />
-  );
-};
+const Image = forwardRef(
+  ({ a11yTitle, fallback, onError, opacity, fill, src, ...rest }, ref) => {
```
Comment (ShimiSun) [inline src/js/components/Image/Image.js:5]: I think image used to work as expected with with `ref`, why is that addition needed?

File: src/js/components/Meter/Bar.js [hunk 8]
```diff
@@ -1,25 +1,15 @@
...
-    round,
-    size,
-    theme,
-    thickness,
-    values,
-    ...rest
-  } = props;
+const Bar = forwardRef((props, ref) => {
+  const { background, max, round, size, thickness, values, ...rest } = props;
+  const theme = useContext(ThemeContext);
```
Comment (ShimiSun) [inline src/js/components/Meter/Bar.js:12]:
```suggestion
  const theme = useContext(ThemeContext) || defaultProps.theme;
```

File: src/js/components/Image/Image.js (hunk 7 above)
Comment (ericsoderberghp) [reply src/js/components/Image/Image.js:5]: Switching from `withForwardRef` HOC to `forwardRef` directly from React.
